spec/support/fixtures/foo

```diff
@@ -463,6 +463,9 @@ module Foo
       end
 
       class SubCommand < Hanami::CLI::Command
+        before { puts 'Before sub command' }
```

Comment:
I'd not add this as DSL for a command.
The point of callbacks feature is to extend somebody else's commands. We have two cases:

  * If you are the author of this command, you're writing this concrete class. There is no need to have a callback as you can implement all the logic in `#call`.
  * You are a person who want to extend this command, you can't access `Subcommand.before`, because that requires to monkey-patch the original class.

What we can do is:

```ruby
# gem `foo`

module Foo
  module CLI
    class Hello < Hanami::Cli::Command
      def call(*)
        puts "Hello"
      end
    end
  end
end

Foo::CLI.register "greet", Foo::CLI::Hello
```

Then in the second gem (`bar`)

```ruby
# gem `bar`
require "foo"

module Bar
  module CLI
    class World < Hanami::Cli::Command
      def call(*)
        puts "world!"
      end
    end
  end
end

Foo::CLI.after "greet", Bar::CLI::World
```

And finally in the shell:

```shell
% greet
Hello
world!
```

Reply:
yep, I'll do it. My next step (after specs and cleanup) - update command resolver to call `before` and `after` methods for each command with new "callback" command